app/controllers/preferences_controller.rb

```diff
@@ -1,8 +1,67 @@
 # frozen_string_literal: true
 
 class PreferencesController < ApplicationController
+
+  # Callbacks
+  before_action :set_preference, only: %i[edit update show destroy] 
+
   def index
-    @preferences = current_user.preferences
+    @preferences = current_user.preferences.includes(:user)
     @pagy, @records = pagy(@preferences)
   end
-end
+
+  def show; end
+
+  def new 
+    @preference = Preference.new 
+  end
+  
+  def edit; end
+
+
+  def create # create a function to create a new preference
+    @preference = current_user.preferences.new(preference_params) # create a new preference for the current user
+
+    if @preference.save  # if the preference is saved successfully
+      redirect_to preferences_path, notice: 'Preference was successfully created.'  # redirect to the preferences page with a success message
+    else
+      render :new, status: :unprocessable_entity  # if the preference is not saved, render the new template with an unprocessable entity status
+    end
+  end
+
+  def update
+    @preference = Preference.find(params[:id])  # find the preference with the given id
```

Comment:
```suggestion
```
This is not needed since the `before_action :set_preference` callback is already loading `@preference`